core/src/main/java/hudson/model/Run.java

```diff
@@ -1697,15 +1709,14 @@ protected final void execute(@Nonnull RunExecution job) {
             long start = System.currentTimeMillis();
 
             try {
+                Computer computer = Computer.currentComputer();
+                if (computer != null) {
+                    setCharset(computer.getDefaultCharset());
```

Comment:
This part is what I think is really wrong. We should change the charset of all builds to be UTF-8, unconditionally, so that miscellaneous messages can always be safely written to the log file; but make sure that anything which copies external process output (namely, `Launcher`) transcodes that output on either the master or remote side.

Cf. my corresponding changes for Pipeline in https://github.com/jenkinsci/workflow-job-plugin/pull/27 and https://github.com/jenkinsci/durable-task-plugin/pull/29, with integration test (`ShellStepTest.encoding`) in https://github.com/jenkinsci/workflow-durable-task-step-plugin/pull/21.

Reply:
I am sure you are right. Ideally only UTF-8 would be used, though I was hesitant to go through and rip out anything relating to charsets. Right now I have changed the implementations of Computer.getDefaultCharset() to return `StandardCharsets.UTF_8`, so effectively this is what is being done, but without removing the scaffolding just yet.